### ISSUE TITLE
The selected 'options' for the combobox are not announced as 'selected' by NVDA

### ISSUE THREAD

**Issue by ashankggupta** [nvaccess/nvda#18224]
Summary
The selected option of the combobox is not announced as "selected" although aria-selected is set to "TRUE" on the element with role="option".

Actual Result:
The selected option of the combobox is not announced as "selected" while the options with aria-selected="false" are also not announced as 'not selected'.

Expected Result:
Ensure that "selected" and "not selected" states are announced for the selected and not selected option of the combobox. 
NOTE: aria-checked attribute is alsi used with the items having aria-selected="true" attribute as a fallback to indicate the selected state.

Additional Information
We have aria-selected and aria-checked attributes to enhance accessibility and provide a consistent user experience across different operating systems.

On Windows, the native select element changes its value as you navigate through options using the arrow keys, without needing to press Enter. However, on Mac, users must press Enter to confirm their selection after navigating with the arrow keys.

Our combobox follows the Mac behavior, requiring users to press Enter to confirm their choice. The aria-selected attribute is used to indicate the currently highlighted option as you navigate through the list. Additionally, a visual check mark is displayed on the selected value, which only moves when Enter is pressed. We use the aria-checked attribute to communicate this visual check mark.

Code Snippet:
`<div id="listbox-id-6" class="slds-dropdown slds-dropdown_length-5 slds-dropdown_fluid" role="listbox" aria-label="{{Placeholder for Dropdown Items}}" tabindex="0" aria-busy="false">
     <ul class="slds-listbox slds-listbox_vertical" role="presentation">
         <li role="presentation" class="slds-listbox__item">
               <div **aria-selected="true" aria-checked="true"** id="option208" class="slds-media slds-listbox__option slds-listbox__option_plain slds-media_small slds-is-selected slds-has-focus" role="option"><span class="slds-media__figure slds-listbox__option-icon"><span class="slds-icon_container slds-icon-utility-check slds-current-color"><svg class="slds-icon slds-icon_x-small" aria-hidden="true"><use xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#check"></use></svg></span></span><span class="slds-media__body"><span class="slds-truncate" title="Accounts">Accounts</span></span></div>
         </li>`

....

Operating System and version
Windows 10

Browser and version:
Chrome 136.0.7103.114 (64-bit)

**Comment by Adriani90** [comment]
It seems aria checked is atleast announced but only when using object navigation. Aria selected is not announced in any browser.
This issue also affects Firefox.
cc: @jcsteh, @SaschaCowley I am not sure whether the code snippet is valid here, could you please have a look?

**Comment by gerald-hartig** [comment]
Can you please provide a codepen example of the issue where we can independently reproduce this.

**Comment by SaschaCowley** [comment]
Closing as requested information has not been provided. Furthermore, this works as expected for me with the following minimal sample:

```
data:text/html,<div role="listbox"><span tabindex=0 role=option aria-selected=true>One</span><span tabindex=0 role=option aria-selected=false>Two</span><span tabindex=0 role="option" aria-selected="false">Three</span></div>
```
